projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1602,10 +1600,50 @@
 					}, 150 );
 				}
 			} );
+			var currentSlideObj = carousel.slides[ options.startIndex ];
+			var currentSlideImage =
+				currentSlideObj && currentSlideObj.el ? currentSlideObj.el.querySelector( 'img' ) : null;
+			var isCurrentSlideImageLoaded =
+				currentSlideImage && currentSlideImage.complete && currentSlideImage.naturalHeight !== 0;
+
+			domUtil.fadeIn( carousel.overlay );
 
-			domUtil.fadeIn( carousel.overlay, function () {
+			if ( isCurrentSlideImageLoaded ) {
+				toggleLoader( false );
+				carousel.container.classList.add( 'jp-carousel-fade-in-image' );
 				domUtil.emitEvent( carousel.overlay, 'jp_carousel.afterOpen' );
-			} );
+				return;
+			}
+
+			toggleLoader( true );
+			// Grab the selected image and preload it before fading it in.
+			var newImage = new Image();
+
+			newImage.onload = function () {
+				toggleLoader( false );
+				carousel.container.classList.add( 'jp-carousel-fade-in-image' );
+				domUtil.emitEvent( carousel.overlay, 'jp_carousel.afterOpen' );
+			};
+			newImage.onerror = function () {
+				toggleLoader( false );
+				carousel.container.classList.add( 'jp-carousel-fade-in-image' );
+				domUtil.emitEvent( carousel.overlay, 'jp_carousel.afterOpen' );
+			};
+			newImage.src = currentSlideObj.attrs.src;
```

Comment:
This line is throwing an error for me. I think the above suggestion to default to the first slide if an index isn't specified should fix it.

![image](https://user-images.githubusercontent.com/14988353/127268844-21776338-e670-4a39-8f8e-78a28842b665.png)

Reply:
Oh, thanks for finding that! I've applied the suggested commit. 🙇

Reply:
Awesome, thanks for that. The code changes look good to me now, and that's working. I'll just do a little more cross browser testing, then give it the ✅